tests/inheritance.test.ts

```diff
@@ -0,0 +1,306 @@
+import { assertPrint, assertFail, assertTCFail, assertTC } from "./asserts.test";
+import { NUM, BOOL, NONE, CLASS } from "./helpers.test"
+
+describe("Inheritance tests", () => {
+  assertPrint("field-read", `
+class A(object):
+  x:int = 1
+  f:int = 1
+      
+class B(A):
+  y:int = 2
+      
+class C(B):
+  z:int = 2
+
+c: C = None
+c = C()
+print(c.x)
+print(c.y)
+print(c.z)
+`, [`1`,'2','2']);
+assertPrint("field-update-read", `
+class A(object):
+  x:int = 1
+  f:int = 1
+      
+class B(A):
+  y:int = 2
+      
+class C(B):
+  z:int = 2
+
+c: C = None
+c = C()
+c.x = 10
+c.y = 11
+c.z = 12
+c.f = 13
+print(c.x)
+print(c.y)
+print(c.z)
+print(c.f)
+`, [`10`,'11','12','13']);
+assertPrint("field-update-read-multiple-classes", `
+class A(object):
+  x:int = 1
+  f:int = 1
+      
+class B(A):
+  y:int = 2
+      
+class C(B):
+  z:int = 2
+
+class D(object):
+  x:int = 10
+      
+class E(D):
+  y:int = 20
+
+c: C = None
+e: E = None
+e = E()
+c = C()
+print(c.x)
+print(c.y)
+print(c.z)
+print(e.x)
+print(e.y)
+`, [`1`,'2','2','10','20']);
+assertPrint("field-no-method-classes", `
+class Single(object):
+	a : int = 1
+
+
+class Two(Single):
+	b : int = 2
+
+class Three(Two):
+	c : int = 3
+
+
+l : Three = None 
+l = Three()
+print(l.c)
+`, [`3`]);
+assertPrint("simple-method-call", `
+class Single(object):
+	a : int = 1
+
+	def sum1(self: Single) -> int: 
+		return self.a
+
+class Two(Single):
+	b : int = 2
+
+	def sum2(self: Two) -> int: 
+		return self.a + self.b
+
+class Three(Two):
+	c : int = 3
+
+	def sum3(self: Three) -> int: 
+		return self.a + self.b + self.c
+
+l : Three = None 
+l = Three()
+print(l.sum3())
+`, [`6`]);
+assertPrint("base-class-method-call", `
+class A(object):
+    x: int = 1
+
+    def increment(self: A, i: int) -> int:
+        return self.x + i
+    
+
+class B(A):
+
+    y:int = 10
+    
+a : A = None
+a = B()
+print(a.increment(1))
+`, [`2`]);
+assertPrint("method-doesnt-exist-in-super-class", `
```

Comment:
This is the same name as another test below.

Reply:
Removed the duplicate test case.